peachjam/adapters/gazettes.py

```diff
@@ -23,13 +23,37 @@ def check_for_updates(self, last_refreshed):
             f"Checking for new gazettes from Gazettes.Africa since {last_refreshed}"
         )
 
-        new_gazettes = (
-            CoreDocument.objects.filter(
-                doc_type="gazette", jurisdiction=self.jurisdiction
+        if "-" not in self.jurisdiction and "*" not in self.jurisdiction:

```

Comment:
You've got a lot of duplication here, which you want to avoid, because it's prone to bugs and more difficult to update. Remember that a queryset can easily have extra filters added to it once it's created.

Instead:

1. split `self.jurisdiction` on "-" and get a juri code and a locality code (default locality to None if no "-")
2. create the queryset and filter the queryset on the juri code
3. check the locality code and filter the queryset accordingly, if necessary